src/00/02-introspect/proposal.adoc

```diff
@@ -0,0 +1,879 @@
+
+--
+We define a tool (producer or consumer) to be _backward compatible_, for
+semantic versioning, with another tool (consumer or producer) when the
+consumer that implements an older version of the API can operate, with the same
+semantics, when interacting with a producer that implements a newer version of
+the API, and vice versa.
+
+For example: If a producer generates JSON structured data. In a newer,
+compatible, version if may decided to introduce a new field. If such a field
+can be ignored by the consumer such that ignoring it does not change the
+operational semantics of the consumer the API would be considered
+_backward compatible_. And hence could be indicated with a MINOR or PATCH
+version difference per semantic versioning.
+
+The specifics of how the API behaves to achieve _backward compatible_ changes
+is up to the individual specification of the capabilities. As the ability to be
+_backward compatible_ varies with the specifics of many factors, like tool
+options, data formats, and so on.
+--
+
+==== Single Version
+
+A single version in this proposal is composed of a one to three dotted whole
+numbers. The numbers are expected to be strictly increasing.
+Following SemVer footnote:SemVer[] a change to the MAJOR version indicates a
+backward incompatible change. And changes to the MINOR and PATCH versions
+indicate backward compatible changes.
+The format for the version must match the regular expression: footnote:Regex[]
+
+[source,plain_text]
+----
+^[0-9]+([.][0-9]+){0,2}$
+----
+
+==== Version Range
+
+A version range in this proposal indicates a lower and upper bound of versions.
+It is composed of a pair of versions, separated by a comma, and bracketed by
+either an inclusive or exclusive symbol. This matches the intuition of a
+mathematic interval, but with the use of the version triplet number line.
+footnote:[Wikipedia: Interval (mathematics) (https://en.wikipedia.org/wiki/Interval_(mathematics))]
+Like the interval notation the `()` brackets indicate an exclusive point. And
+the `[]` brackets indicate an inclusive point. As versions are decidedly not
+single integers we use a `,` (comma) to separate the start and end of the range
+instead of using `..`. Hence the format for the version range must match the
+regular expression:
+footnote:Regex[]
+
+[source,plain_text]
+----
+^[[(][0-9]+([.][0-9]+){0,2},[0-9]+([.][0-9]+){0,2}[)\\]]$
+----
+
+==== Multiple Ranges
+
+There are situations where specifying only one version range for what the
+application supports is not sufficient. For example an application may decide
+that they add support for a `2.0.0` version but not support further `1.x.y`
+versions. In that case it's important to be precise in informing consumers of
+this fact. To allow for that situation one can specify a JSON array instead of
+the single JSON string for the version range. For example:
+
+[source,plain_text]
+----
+include::example-07.json[]
+----
+
+=== Version Matching
```

Comment:
The RPM, PEP440, and Debian folks have all done work in this space. We should look at and reference their pre-existing work.

If the concern here is "how do we handle ranges and compatibility?" that is not our job, and there are solutions for it such as [pubgrub](https://nex3.medium.com/pubgrub-2fb6470504f), which has been adopted by Dart.